### PR TITLE
CI: update GH Actions for sanity-test

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: test/sanity/run-test.sh [hunk 1]
```diff
@@ -32,6 +32,12 @@ function install_csi_sanity_bin {
   popd
 }
 
+function provision_nfs_server {
```
Comment (andyzhangx) [inline test/sanity/run-test.sh:35]: move this step to github action only.

Comment (mayankshah1607) [reply test/sanity/run-test.sh:35]: if we move this to GH actions workflow, we may not be able to reuse it for prow

Comment (mayankshah1607) [reply test/sanity/run-test.sh:35]: can we leave this here while we're experimenting with DinD for Prow?

Comment (andyzhangx) [reply test/sanity/run-test.sh:35]: ok, keep it.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
/release-note-none

**Pull request merged**